From 628b90cb3b72d3c2e0a3514833c227712460a3b3 Mon Sep 17 00:00:00 2001
From: Larry Woodman <lwoodman@redhat.com>
Date: Thu, 25 Sep 2014 17:38:21 -0400
Subject: [kernel] sched/numa: Initialize new idle_balance stats in sd_numa_init()

Message-id: <1411666702-16932-2-git-send-email-lwoodman@redhat.com>
Patchwork-id: 96727
O-Subject: [RHEL7.1 PATCH 1/2] sched/numa: Initialize new idle_balance stats in sd_numa_init()
Bugzilla: 1103828
RH-Acked-by: Rik van Riel <riel@redhat.com>

commit 2b4cfe64dee0d84506b951d81bf55d9891744d25
 Author: Jason Low <jason.low2@hp.com>
 Date:   Wed Apr 23 18:30:34 2014 -0700

    sched/numa: Initialize newidle balance stats in sd_numa_init()

    Also initialize the per-sd variables for newidle load balancing
    in sd_numa_init().

    Signed-off-by: Jason Low <jason.low2@hp.com>
    Acked-by: morten.rasmussen@arm.com
    Cc: daniel.lezcano@linaro.org
    Cc: alex.shi@linaro.org
    Cc: preeti@linux.vnet.ibm.com
    Cc: efault@gmx.de
    Cc: vincent.guittot@linaro.org
    Cc: aswin@hp.com
    Cc: chegu_vinod@hp.com
    Signed-off-by: Peter Zijlstra <peterz@infradead.org>
    Link: http://lkml.kernel.org/r/1398303035-18255-3-git-send-email-jason.low2@hp.com
    Signed-off-by: Ingo Molnar <mingo@kernel.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index 8a2b02e..577957b 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -6455,6 +6455,8 @@ sd_numa_init(struct sched_domain_topology_level *tl, int cpu)
      ,
   .last_balance  = jiffies,
   .balance_interval = sd_weight,
+  .max_newidle_lb_cost = 0,
+  .next_decay_max_lb_cost = jiffies,
  };
  SD_INIT_NAME(sd, NUMA);
  sd->private = &tl->data;
-- 
1.7.1